src/porepy/numerics/ad/operators.py

```diff
@@ -1431,6 +1465,20 @@ def shape(self) -> tuple[int, int]:
         """Shape of the wrapped matrix."""
         return self._shape
 
+    @staticmethod
+    def _compute_spmatrix_hash(mat: sps.spmatrix) -> str:
+        """Utility function that handles all the sparse formats to compute the hash."""
+        # proper handling of all the formats like csr, coo, etc..
+        if isinstance(mat, (sps.csr_matrix, sps.csc_matrix, sps.bsr_matrix)):
```

Comment:
Scipy encourages gradual introduction of csr_array etc., see https://docs.scipy.org/doc/scipy/reference/sparse.html, which will break this logic. Please account for arrays as well as matrices.

Reply:
Done